[CS2113-T18-2] FinancialPlanner

ryan1604 opened the pull request:
Financial Planner helps individuals manage their finances effectively and achieve their financial goals. The purpose of such an application is to provide users with a range of tools and features to help them better understand their financial situation.

maanyos commented on src/main/java/seedu/financialplanner/commands/AbstractCommand.java:
```diff
@@ -0,0 +1,5 @@
+package seedu.financialplanner.commands;
+
+public abstract class AbstractCommand {
```
no need to put abstract in the class name

maanyos commented on src/main/java/seedu/financialplanner/storage/LoadData.java:
```diff
@@ -0,0 +1,96 @@
+        }
+    }
+
+    private static void loadBudget(String[] split) {
+        double initial = Double.parseDouble(split[1].trim());
+        double current = Double.parseDouble(split[2].trim());
+        Budget.load(initial, current);
+    }
```
might want to add some safeguards here

maanyos commented on src/main/java/seedu/financialplanner/storage/LoadData.java:
```diff
@@ -0,0 +1,96 @@
+                String[] split = line.split("\\|");
+                String type = split[0].trim();
+                switch (type) {
+                case "I":
```
is this supposed to be empty?

maanyos commented on src/main/java/seedu/financialplanner/commands/BudgetCommand.java:
```diff
@@ -0,0 +1,89 @@
+            throw new FinancialPlannerException("There is an existing budget, did you mean update?");
+        }
+
+        if (command.equals("update") && !Budget.hasBudget()) {
+            logger.log(Level.WARNING, "Invalid command: Trying to update non-existent budget");
+            throw new FinancialPlannerException("There is no budget set yet, did you mean set?");
+        }
+
+        if (!rawCommand.extraArgs.containsKey("b")) {
+            logger.log(Level.WARNING, "Missing argument /b in command");
+            throw new IllegalArgumentException("Missing /b argument.");
+        }
+
+        try {
+            logger.log(Level.INFO, "Parsing budget as double");
+            budget = Double.parseDouble(rawCommand.extraArgs.get("b"));
+        } catch (IllegalArgumentException e) {
+            logger.log(Level.WARNING, "Invalid value for budget");
+            throw new IllegalArgumentException("Budget must be a number.");
+        }
+
+        if (budget <= 0) {
+            logger.log(Level.WARNING, "Invalid value for budget.");
+            throw new FinancialPlannerException("Budget should be greater than 0.");
+        }
+
+        if (budget > Cashflow.getBalance()) {
+            logger.log(Level.WARNING, "Invalid value for budget");
+            throw new FinancialPlannerException("Budget should be lower than total balance.");
+        }
+
+        assert budget > 0 && budget <= Cashflow.getBalance() : "Budget should be greater than 0 and less than " +
+                "or equal to total balance";
+        rawCommand.extraArgs.remove("b");
+
+        if (!rawCommand.extraArgs.isEmpty()) {
+            String unknownExtraArgument = new ArrayList<>(rawCommand.extraArgs.keySet()).get(0);
+            throw new IllegalArgumentException(String.format("Unknown extra argument: %s", unknownExtraArgument));
+        }
+    }
```
try using a neater way to put safeguards

maanyos commented on src/main/java/seedu/financialplanner/list/CashflowList.java:
```diff
@@ -0,0 +1,128 @@
+public class CashflowList {
+    public static Logger logger = Logger.getLogger("Financial Planner Logger");
+
+    public static final CashflowList INSTANCE = new CashflowList();
+    public final ArrayList<Cashflow> list = new ArrayList<>();
```
use better names for these lists

maanyos commented on src/main/java/seedu/financialplanner/commands/RawCommand.java:
```diff
@@ -0,0 +1,28 @@
+import java.util.List;
+import java.util.Map;
+
+public class RawCommand {
```
how does this class relate to the other commands?

maanyos commented on src/main/java/seedu/financialplanner/commands/FindCommand.java:
```diff
@@ -0,0 +1,56 @@
+        WatchList watchList = WatchList.INSTANCE;
+        ArrayList<String> foundedFinancialList = new ArrayList<>();
+        ArrayList<String> foundedWatchList = new ArrayList<>();
+        for (int i = 0; i < cashflowList.list.size(); i++) {
+            if (cashflowList.list.get(i).toString().contains(description)) {
+                String output = cashflowList.list.get(i).toString()+" | Index: "+(i+1);
+                foundedFinancialList.add(output);
+            }
+        }
```
can also just iterate through the objects instead of using their indexes

maanyos reviewed (commented):
Good use of loggers and assertions.

I think you tried to apply the singleton principle for some classes. It will be covered in one of the lectures, you can also read ahead under week 10's topics. The current implementation can improve a lot, particularly the instance names, and the way they are accessed.

Commit:
Change method name for clarity

Commit:
Update DG with recur feature

Commit:
Add recur class and sequence diagrams

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

Commit:
Merge branch 'master' into Branch-UpdateDG

hshiah committed:
Solve suggestions of Minwei

Commit:
Update DG content page

Commit:
Update UG explaining the meaning of cashflow

hshiah committed:
Solve minor errors.

hshiah committed:
Merge pull request #134 from hshiah/hshiah

Delete Reminder and display reminder

wwweert123 committed:
Add some fix to DG and UG

wwweert123 committed:
Merge branch 'master' into VisDG

woodenclock commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
 # Developer Guide
 
+## Table of contents
```
Good use of indentation to organize the Table of Contents.
However, since there are 3 types of indentation, could there be a better way of labelling each line? say 1.1.1?
Just a small suggestion.

woodenclock commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+  * [Glossary](#glossary)
+  * [Instructions for manual testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
```
Comprehensive acknowledgement of others.
Properly citing third-party libraries and tools ensures transparency and gives credit to the original creators. 
This section showcases the developer's integrity and understanding of open-source norms.

woodenclock commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
```
Perhaps it would be good to have a Architecture Diagram showcasing your overall design and mechanism of FinancialPlanner. This would give the Developer their needed context should they want to further implement other features to integrate into your program

woodenclock commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+  * Advantage: Changes are saved after every command.
+  * Disadvantage: Executing command might slow down the program when there is a large amount of data to be saved.
+
+### Visualization Feature 
```
Kudos for integrating Xchart!
By having the visual aids, it would definitely allow the user to be aware of their expenses at a glance.
Even though FinancialPlanner is CLI, I like the GUI component added to enhance the UI/UX

woodenclock commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Storage Component
```
The relationship between Storage and SaveData/LoadData seems slightly confusing. 
Typically, if Storage is a class providing functionalities like saving and loading, it may have methods for those, rather than being connected to abstract classes representing those actions. 
Consider redefining these relationships for clarity.

A clearer distinction or some elaboration on what differentiates Cashflow and Budget would be beneficial.

woodenclock commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+According to the chart type (Pie/Bar) argument and the Hashmap obtained from the categorizer passed in, 
+the visualizer displays the specified visualization chart by calling the charting library Xchart.
+
+### Class Diagram
```
Visualizer seems to have methods that return some kind of chart, but the return types are not indicated. 
It would be clearer if return types were included.

The relationship between Categorizer and other classes is not immediately evident. 
Perhaps some connecting lines or indications of method calls/dependencies would be helpful.

wwweert123 replied:
VIsualizer would do the showing of chart itself. Thus it is a void method call to visualizer to print the chart. 

Thanks for the feedback on the class diagram!
Will work on that!

woodenclock commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+
+![](images/vis/visualisationClass.png)
+
+### Sequence Diagram 
```
In the sequence diagram, it would be beneficial to include return arrows, especially for method calls that yield some kind of result.
The method printDisplayChart(type) is invoked on Ui from VisCommand, but from the class diagram, it seems like the method belongs to the Visualizer class. This discrepancy should be resolved for consistency.
Or am I just confusing myself?

wwweert123 replied:
Thanks for the feedback! I can see the confusion. printDisplayChart prints a message telling the user that a chart is being displayed. Visualizer has a method for displaying the actual chart. I have renamed the method for UI to make it clearer. I will add the return arrows as well!

woodenclock reviewed (commented):
Generally good job in making a comprehensive developer guide.
I really like your visualizer, that enhances the UI/UX.
Diagrams were present to support the understanding of the program

danielpappa commented on docs/DeveloperGuide.md:
Overall very good DG, very precise and in depth analysis of the code and detailed description.
Maybe could use a single and coherent way of using bullet points (right now the DG contains >>, 1), 1. and Step 1). This could make the DG look as if actually written by a single person.

danielpappa commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
 ## Acknowledgements
 
+**Xchart (A Simple Charting Library for Java)**
+- author: KNOWN
```
Maybe there is a typo here: shouldn't it be `author: KNOWM`?

wwweert123 replied:
THanks for spotting!

danielpappa commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```
Maybe you could add a brief introduction and overview of your Financial Planner, explaining the goal and purpose as well as its rough overall design in form of an arbitrary architecture diagram. This would also allow further developers an immediate insight into your product and align them with your ideas and implementations from the start.

danielpappa commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+* Saving the data once upon exit (Currently implemented):
+  * Advantage: Better efficiency and performance of the program.
+  * Disadvantage: If the program crashes or exits incorrectly, data will not be saved.
+* Saving the data after every command:
```
Perhaps it should be specified whether this way is currently implemented or not (as is done for the other option) to keep the considerations coherent. Right now I assume it is not, but I would still not be sure.

ChoonSiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+
+Overall 
+
+![](images/vis/visualisationSequence.png)
```
Should there be a <\<class\>> above Categorizer and Visualizer?

danielpappa replied:
Maybe you could include return arrows. Also it could perhaps be helpful to make the alt and opt boxes coherent by applying the coloring to all sequence diagrams in the DG (helps create impression that DG was written by a single person).

danielpappa commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+Output
+
+`Displaying piechart for expense`
+A message will be shown telling you that the chart is being displayed
```
Really nice implementation. This feature is straightforward and meets the user requirements.

ChoonSiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+
+#### Diagrams
+Given below is the class diagram showing the class structure of the add income/expense mechanism:
+![](images/CashflowClassDiagram.png)
```
Perhaps you will want to refer to the PlantUML docs to remove the C and E circles to comply with our standard of UML diagrams.

ChoonSiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+According to the cashflow type (Income/Expense) arugment passed in, the Categorizer sorts the 
+specified cashflow entry according to type using a Hashmap which is returned and used by the Visualizer
+
+Visualizer's Role: 
```
Nice use charts for user to visualize their income/expenses.

danielpappa commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+
+### Class Diagram
+
+![](images/vis/visualisationClass.png)
```
In the text above Command and RawCommand are omitted, whereas it seems like there should be a dependency between the Categorizer and VisCommand. Maybe some more detailed steps could help clarify these doubts.

wwweert123 replied:
Thanks for your feedback, will add it in!

ChoonSiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+![](images/CashflowClassDiagram.png)
+
+Given below is the sequence diagram showing the add income/expense mechanism:
+![](images/AddCashflowSequence.png)
```
Activation Bar is missing in this diagram. May want to refer to the Sequence Diagram examples in the CS2113 Website.

danielpappa reviewed (commented):
Overall great DG! Very detailed description and plenty of diagrams and code snippets to transmit the content in an intuitive way. Maybe could try to make it more coherent by paying attention to how each detail is actually formatted (colors, bullet points etc.). Kudos also for the use of diagrams!

ChoonSiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+
+This feature has 5 functions, `set`, `update`, `delete`, `reset`, and `view`.
+
+![](images/Budget.png)
```
Nice use of reference frames for neater view of the diagram.

ChoonSiang reviewed (commented):
Overall very nice use of diagrams, except some minor bugs in the UML diagrams

ChoonSiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,249 @@
+
 
 ## Product scope
 ### Target user profile
```
May want to add more User Stories to accurately reflect your actual feature implementation in each version.

ChoonSiang reviewed (commented):
Add on to previous comment

Commit:
Update DG

Commit:
Fix typo to be in camelCase

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

Commit:
Merge branch 'master' into Branch-UpdateDG

NeoMinWei committed:
Merge pull request #135 from NeoMinWei/Branch-UpdateDG

Branch update dg

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

Commit:
FIx formatting error in UG

NeoMinWei committed:
Merge pull request #140 from NeoMinWei/Branch-UGquickfix

Branch ug quickfix

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

hshiah committed:
Finish all user stories for v2.0

YFshadaow committed:
Add help command

YFshadaow committed:
Update EXPECTED.TXT for UI testing

hshiah committed:
Update src/main/java/seedu/financialplanner/commands/MarkGoalCommand.java

Co-authored-by: Neo Min Wei <110591123+NeoMinWei@users.noreply.github.com>

hshiah committed:
Update src/main/java/seedu/financialplanner/commands/MarkGoalCommand.java

Co-authored-by: Neo Min Wei <110591123+NeoMinWei@users.noreply.github.com>

hshiah committed:
Update ExpenseType.java

hshiah committed:
Update MarkGoalCommand.java

hshiah committed:
Update src/main/java/seedu/financialplanner/commands/DeleteGoalCommand.java

Co-authored-by: ryan <94791461+ryan1604@users.noreply.github.com>

hshiah committed:
Merge pull request #143 from hshiah/temp

Finish all user stories for v2.0

wwweert123 committed:
Merge pull request #141 from wwweert123/VisDG

Add to DG and fix bugs

YFshadaow committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp into help-command

# Conflicts:
#	src/main/java/seedu/financialplanner/utils/Parser.java

YFshadaow committed:
Resolve merge conflicts

YFshadaow committed:
Merge pull request #142 from YFshadaow/help-command

Add help command

janelleenqi commented on docs/DeveloperGuide.md:
````diff
@@ -1,13 +1,304 @@
+
+Example:
+```
+switch (category) {
+        case INCOME:
+            cashflowList.addIncome(amount, incomeType, recur);
+            break;
+        case EXPENSE:
+            cashflowList.addExpense(amount, expenseType, recur);
+            break;
+        default:
+            ui.showMessage("Unidentified entry.");
+            break;
+        }
+```
````
Showing this using a diagram would be better, instead of showing this segment of your code.

janelleenqi replied:
Showing this using a diagram would be better, instead of showing this segment of your code.

SebasFok commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,304 @@
+
+#### Design considerations:
+
+* There are 2 main ways to implement the storage, one is to save the data after every command, and the other is to save 
```
Great job explaining the pros and cons of the design consideration. Perhaps you could consider explaining to the reader why you chose to implement it this way over the other instead of just stating what you chose.

janelleenqi commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,304 @@
+Example: `budget view`
+
 
 ## Product scope
 ### Target user profile
```
You may want to define your product scope and outline your target user profile. This will help make your developer guide more comprehensive.

SebasFok replied:
Perhaps you could have a clear target user profile which could give the reader a better picture of who the code should cater for

janelleenqi commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,304 @@
+initial and current budget. This is done through `updateBudget(500)` method in `Budget.java`. Both functions can be seen 
+in the diagram above
+
+#### Delete budget:
+
+![](images/deleteBudget.png)
+
+The budget will be deleted by setting the initial and current budget to 0 through the `deleteBudget()` method in `Budget.java`.
+
+Example: `budget delete`
+
+#### Reset budget:
+
+![](images/resetBudget.png)
+
+The budget will be reset by resetting the current budget to the initial budget through the `resetBudget()` method in `Budget.java`.
+
+Example: `budget reset`
+
+#### View budget:
+
+![](images/viewBudget.png)
+
+The current budget will be shown to the user through the `Ui`.
+
+Example: `budget view`
```
Good job providing sequence diagrams for 3 functions of your budget feature!

janelleenqi commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,304 @@
+
+Below are the steps that shows the implementation of the recurring cashflow feature.
+
+#### Step 1
```
I think it would be good to add a summary of each step
`#### Step 1: Store date of cashflow added`

janelleenqi commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,304 @@
+
+#### Add Income/Expense Class Diagram
+Given below is the class diagram showing the class structure of the add income/expense mechanism:
+![](images/cashflow/CashflowClassDiagram.png)
```
Good job adding class diagrams!

janelleenqi reviewed (commented):
Overall, good job! You have your design and implementation more or less completed! Good use of diagrams!

SebasFok commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,304 @@
+Given below is the class diagram showing the class structure of the add income/expense mechanism:
+![](images/cashflow/CashflowClassDiagram.png)
+#### Add Income/Expense Sequence Diagram
+Given below is the sequence diagram showing the add income/expense mechanism:
```
Very clear sequence diagram that has good amount of white space to make it readable. Gives a good picture of how the add income/expense works and when each of them are called

SebasFok commented on docs/DeveloperGuide.md:
```diff
@@ -1,13 +1,304 @@
+
+![](images/vis/visualisationSequence.png)
+
+Categorizer
```
Great use of ref frames to show the overall sequence diagram. Could maybe add return arrows to make the diagram slightly clearer.

wwweert123 committed:
Fix DG sequence diagram for cat and vis

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

hshiah committed:
Update UG. Fix some minor errors of reminder and goal. Update the saving and loading for goals and reminder.

wwweert123 committed:
Fix code according to DG

hshiah committed:
Update WishList.java

hshiah committed:
Update ReminderList.java

hshiah committed:
Merge pull request #151 from hshiah/hshiah

Update UG and storage

wwweert123 committed:
Merge branch 'master' into VisDG

# Conflicts:
#	src/main/java/seedu/financialplanner/commands/VisCommand.java

wwweert123 committed:
Merge pull request #150 from wwweert123/VisDG

Fix DG sequence diagram for cat and vis

Commit:
Remove long method parameters in recur sequence and class diagrams

Commit:
Resolve spacing issues in headers

Commit:
Add description for the types accepted in add income and expense

ryan1604 committed:
Fix bugs

Commit:
Add maximum/minimum value exception

Commit:
Add trimming of arguments from input

Commit:
Add listing of total balance in list command and removing of extra spaces in user input

ryan1604 committed:
Update diagrams, DG and UG

wwweert123 committed:
Add change for loading watchlist with more than 5 stocks, initallize stocks as null

wwweert123 committed:
Merge branch 'master' into FixBugsEnhanceDGUG

ryan1604 committed:
Update UG

ryan1604 committed:
Update test file

wwweert123 committed:
Create new constructor for adding base stocks

ryan1604 committed:
Update reset budget diagram

ryan1604 committed:
Fix typo

wwweert123 committed:
Add mechanism to clean up watchlist as much as possible to remove nonsense entries

wwweert123 committed:
Add to UG to warn user

wwweert123 committed:
Add check for whether null stocks are passed before printing

wwweert123 committed:
Add saving of watchlist after watchlist command

wwweert123 committed:
Add attempt of removing of both normal stock and its uppercase form

wwweert123 committed:
Add details to UG

wwweert123 committed:
Add printing of acknowledgement to APIs

ryan1604 committed:
Add PPP to be completed

wwweert123 committed:
Add instructions for adding stock using JSON

wwweert123 committed:
Add additonal checks for adding stock and handle exception for stock constructor properly

wwweert123 committed:
Add to UG q and A

Commit:
Fix bug

wwweert123 committed:
Fix checkstyle

wwweert123 committed:
Fix code according to reviewer

wwweert123 committed:
Fix typo

ryan1604 committed:
Add return in Budget

ryan1604 committed:
Merge pull request #191 from ryan1604/branch-bugfix

Fix bugs and add improvements according to PE-D

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

Commit:
Merge branch 'master' into Branch-PEBugFix

# Conflicts:
#	src/main/java/seedu/financialplanner/utils/Ui.java

Commit:
Make changes according to PR reviews

wwweert123 committed:
Add trimming for commands

wwweert123 committed:
Add padding to watchlist command

wwweert123 committed:
Add on to UG

wwweert123 committed:
Add PPP

wwweert123 committed:
Merge branch 'master' into FixBugsEnhanceDGUG

# Conflicts:
#	src/main/java/seedu/financialplanner/investments/WatchList.java

wwweert123 committed:
Change colour of message printed

wwweert123 committed:
Merge pull request #193 from wwweert123/FixBugsEnhanceDGUG

Fix bugs enhance dgug

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

Commit:
Merge branch 'master' into Branch-PEBugFix

Commit:
Add PPP

Commit:
Rename PPP file to be in lower case

NeoMinWei committed:
Merge pull request #192 from NeoMinWei/Branch-PEBugFix

Branch pe bug fix

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

ryan1604 committed:
Remove activation bars for ref sd

ryan1604 committed:
Merge branch 'master' into branch-bugfix

ryan1604 committed:
Update overview command

Commit:
Add JUnit testing for DeleteCashflowCommand

Commit:
Add JUnit test for Cashflow

Commit:
Update JUnit tests in CashflowList and AddCashflowCommand to include exception testing

Commit:
Add fullstops to exception messages

Commit:
Fix JUnit

Commit:
Rename method to follow camelCase

Commit:
Add javadoc comments

Commit:
Update text ui test

Commit:
Add fullstop to error message

YFshadaow committed:
Refactor help command structure:
  - Separate command name and usage into each command class
  - Move Command.java, RawCommand.java to new utils package
  - Refactor parser

ryan1604 committed:
Update DG and UG

ryan1604 committed:
Fix bug

ryan1604 committed:
Update Architecture Diagram and fix typo

NeoMinWei committed:
Merge pull request #194 from NeoMinWei/Branch-FinalJunit

Branch final junit

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

ryan1604 committed:
Merge pull request #196 from ryan1604/branch-bugfix

Update UG/DG

wwweert123 committed:
Add javadoc to classes and methods

wwweert123 committed:
Add DG value proposition and user profile

wwweert123 committed:
Add user stories

wwweert123 committed:
Add term to glossary

wwweert123 committed:
Add checkstyle fixes

wwweert123 committed:
Add simple manual testing and fix bugs

wwweert123 committed:
Update contributions for PPP

wwweert123 committed:
Merge pull request #197 from wwweert123/javadocNDG

Javadoc ndg

YFshadaow committed:
Merge branch 'master' of https://github.com/YFshadaow/tp into help-command-refactor

# Conflicts:
#	src/main/java/seedu/financialplanner/commands/AddCashflowCommand.java
#	src/main/java/seedu/financialplanner/commands/AddStockCommand.java
#	src/main/java/seedu/financialplanner/commands/DeleteCashflowCommand.java
#	src/main/java/seedu/financialplanner/commands/DeleteStockCommand.java
#	src/main/java/seedu/financialplanner/commands/OverviewCommand.java
#	src/main/java/seedu/financialplanner/commands/VisCommand.java
#	src/main/java/seedu/financialplanner/commands/WatchListCommand.java

YFshadaow committed:
Resolve merge conflicts

YFshadaow committed:
Finish all command usage and example usage. Add example usage to help messages.

YFshadaow committed:
Remove comment from build.gradle

hshiah committed:
Fix bugs

YFshadaow committed:
Add find and help commands to user guide

YFshadaow committed:
Fix spacing in BudgetCommand.java

YFshadaow committed:
Fix usage of add cashflow command

YFshadaow committed:
Fix incorrect exception message due to invocation target exception casued by reflection usage.

hshiah committed:
Enhancement of reminder

Commit:
Add list command to UG

hshiah committed:
Fix date format

Commit:
Fix typo

Commit:
Fix incorrect directory in UG

YFshadaow committed:
Merge pull request #200 from YFshadaow/help-command-refactor-and-bug-fixes

Help command refactor and bug fixes

hshiah committed:
Merge branch 'master' into hshiah

hshiah committed:
Merge pull request #198 from hshiah/hshiah

Fix bugs

ryan1604 committed:
Update readme and aboutus

hshiah committed:
Fix bugs of loading and saving reminder

hshiah committed:
Fix coding style

hshiah committed:
Merge pull request #203 from hshiah/hshiah

Fix bugs of loading and saving reminder

ryan1604 committed:
Merge pull request #202 from ryan1604/branch-ug

Quick updates

wwweert123 committed:
Fix mistakes

ryan1604 committed:
Merge pull request #201 from NeoMinWei/Branch-FinalUG

Branch final ug

YFshadaow committed:
Solve slf4j binding warning

YFshadaow committed:
Merge pull request #204 from YFshadaow/fix-logger-binding

Solve slf4j binding warning

ryan1604 committed:
Fix UG and DG

ryan1604 committed:
Merge pull request #205 from ryan1604/branch-bugfix

Fix UG and DG

hshiah committed:
Fix bugs

wwweert123 committed:
Fix addstock

wwweert123 committed:
Merge pull request #207 from wwweert123/javadocNDG

Javadoc ndg

hshiah committed:
Update UG

hshiah committed:
Merge pull request #206 from hshiah/hshiah

Fix bugs

wwweert123 committed:
Add mechanism to tell user that he has input the values wrongly in json file

wwweert123 committed:
Add hashing of values to ensure nobody tampers with the json file

wwweert123 committed:
Update UG

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

Commit:
Fix date format in reminder

Commit:
Add exception thrown when impossible dates are detected

wwweert123 committed:
Convert parameter to lowercase for categorizer and visualizer

wwweert123 committed:
Change load watchlist to take take file path as parameter

wwweert123 committed:
Convert to uppercase when passing stock code into Stock constructor

wwweert123 committed:
Add junit test for watchlist

wwweert123 committed:
Add junit test for categorizer and visualizer

wwweert123 committed:
Add return value for javadoc

wwweert123 committed:
Fix typo

Commit:
Update PPP

wwweert123 committed:
Merge branch 'master' into javadocNDG

# Conflicts:
#	docs/UserGuide.md

Commit:
Fix bug

wwweert123 committed:
Fix checkstyle

wwweert123 committed:
Add new line

wwweert123 committed:
Disable test on linx

wwweert123 committed:
Merge pull request #208 from wwweert123/javadocNDG

Javadoc ndg

ryan1604 committed:
Update UG

wwweert123 committed:
Add mechanism to disallow adding stock with other fields

wwweert123 committed:
Add setting of fetch time even for invalid stocks

ryan1604 committed:
Reorder UG

Commit:
Fix displaying of wrong balance when recurring cashflows are added

Commit:
Add error handling for recur value exceeding maximum int value

Commit:
Update user guide

Commit:
Add error handling for index exceeding max int value

Commit:
Update Junit

ryan1604 committed:
Delete note for income type in UG

wwweert123 committed:
Last Changes

wwweert123 committed:
Fix Vis diagrams

wwweert123 committed:
Add watchlist feature to DG

wwweert123 committed:
Add to table of contents

ryan1604 committed:
Merge pull request #210 from ryan1604/branch-bugfix

Update UG

wwweert123 committed:
Update PPP

hshiah committed:
Fix bugs

hshiah committed:
Fix checkstyle

wwweert123 committed:
Update Diagrams

hshiah committed:
Update javadoc

hshiah committed:
Fix checkstyle

ryan1604 committed:
Merge pull request #209 from NeoMinWei/Branch-ReminderDate

Branch reminder date

Commit:
Add activation bars to sequence diagrams

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp into Branch-ReminderDate

wwweert123 committed:
Fix table of content

wwweert123 committed:
Add activation diagram

wwweert123 committed:
Change back

wwweert123 committed:
Merge pull request #211 from wwweert123/javadocNDG

Javadoc ndg

wwweert123 committed:
Add manual testing

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp into Branch-ReminderDate

wwweert123 committed:
Fix typo

wwweert123 committed:
Merge pull request #214 from wwweert123/javadocNDG

Add manual testing

Commit:
Add manual testing

Commit:
Update manual testing

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp into Branch-ReminderDate

# Conflicts:
#	docs/DeveloperGuide.md

ryan1604 committed:
Add manual testing

hshiah committed:
Finished test and fix bugs.

NeoMinWei committed:
Merge pull request #213 from NeoMinWei/Branch-ReminderDate

Add activation bars to sequence diagrams

ryan1604 committed:
Merge branch 'master' into branch-bugfix

# Conflicts:
#	docs/DeveloperGuide.md

ryan1604 committed:
Merge pull request #215 from ryan1604/branch-bugfix

Add manual testing

hshiah committed:
Final update

hshiah committed:
Merge branch 'master' into hshiah

hshiah committed:
Update SetGoalCommandTest.java

hshiah committed:
Update AddReminderCommandTest.java

hshiah committed:
Update ReminderListCommand.java

hshiah committed:
Merge pull request #212 from hshiah/hshiah

Fix bugs

ryan1604 committed:
Fix bug in UGDG

ryan1604 committed:
Update PPP

wwweert123 committed:
Fix broken link

wwweert123 committed:
Add fix that you can only add stocks in capital letters using file

ryan1604 committed:
Merge pull request #216 from ryan1604/branch-bugfix

Fix bug in UGDG

wwweert123 committed:
Merge pull request #217 from wwweert123/changesDG

Changes dg

wwweert123 committed:
Fix broken link

wwweert123 committed:
Merge pull request #218 from wwweert123/changesDG

Fix broken link

wwweert123 committed:
Add about us

wwweert123 committed:
Merge pull request #219 from wwweert123/changesDG

Add about us

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

wwweert123 committed:
Fix DG and UG

wwweert123 committed:
Fix

wwweert123 committed:
fix typo

wwweert123 committed:
Merge pull request #220 from wwweert123/changesDG

Changes dg

ryan1604 committed:
Fix bug in UG

ryan1604 committed:
Merge pull request #221 from ryan1604/branch-bugfix

Fix bug in UG

wwweert123 committed:
Add fix

Commit:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-2/tp

wwweert123 committed:
Merge pull request #222 from wwweert123/changesDG

Commit:
Update add cashflow explanations

NeoMinWei committed:
Merge pull request #223 from NeoMinWei/Branch-DGQuickFix

Branch dg quick fix

ryan1604 committed:
Fix bug

ryan1604 committed:
Fix bug

ryan1604 committed:
Merge pull request #224 from ryan1604/branch-bugfix

Fix bug

wwweert123 committed:
Final Final FInal Final changes

wwweert123 committed:
Merge pull request #225 from wwweert123/changesDG

Final Final FInal Final changes

YFshadaow committed:
Add project portfolio page

YFshadaow committed:
Merge pull request #226 from YFshadaow/yfshadaow-project-portfolio-page

Add project portfolio page

YFshadaow committed:
Update aboutus page

YFshadaow committed:
Merge pull request #227 from YFshadaow/yfshadaow-project-portfolio-page

Update aboutus page

YFshadaow committed:
Remove the non-existing John Doe from AboutUs.md

YFshadaow committed:
Merge pull request #228 from YFshadaow/yfshadaow-project-portfolio-page

Remove the non-existing John Doe from AboutUs.md
